dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/SharingController.java

```diff
@@ -344,6 +350,8 @@ public WebMessage postSharing(
           (Visualization) object,
           sharing.getObject().getUserAccesses(),
           sharing.getObject().getUserGroupAccesses());
+    } else if (object instanceof CategoryOption) {
+      eventPublisher.publishEvent(new CacheInvalidationEvent(this, Region.canDataWriteCocCache));
```

Comment:
Generally, source parameter in ApplicationEvent is supposed to reflect "where the event got published from". It could give some traceability context. 
However, in our case, we don't use that source/context in the listeners (yet). But I kept it in as the recommended good practice. 

Also ApplicationCacheClearedEvent simply hardcodes a string as the source. Replacing it with the "object" parameter moves us further from both best practice and convenience. 
My options would be
1. Keep the controller as the object source since that is where the event got published. (This is the recommended way)
2. Hide the parameter from constructor method and hardcode something inside the constructor. Similar to what we have done in ApplicationCacheClearedEvent.java. (This is mainly convenience over best practice)

What do you think?

Reply:
I would go with 1 if this is the good practice, I was just asking as it seemed strange.

About the invalidate one record I was asking if it was possible to do that in this specific case. Can you invalidate just the record for the current COC?